pydis_site/apps/api/viewsets/bot/infraction.py

```diff
@@ -28,11 +28,12 @@ class InfractionViewSet(CreateModelMixin, RetrieveModelMixin, ListModelMixin, Ge
 
     #### Query parameters
     - **active** `bool`: whether the infraction is still active
-    - **actor** `int`: snowflake of the user which applied the infraction
+    - **actor__id** `int`: snowflake of the user which applied the infraction
```

Comment:
Do you have a test for that? I can't find it, but when I test it, providing `user` as a param is ingored, while `user__id` works:

Full infraction table:
```json
Status code: 200
[
    {
        "id": 1,
        "inserted_at": "2019-04-27T20:54:53.930875Z",
        "expires_at": null,
        "active": false,
        "user": 336843820513755157,
        "actor": 336843820513755157,
        "type": "watch",
        "reason": "Watch that man",
        "hidden": true
    },
    {
        "id": 2,
        "inserted_at": "2019-04-27T21:19:36.568151Z",
        "expires_at": null,
        "active": false,
        "user": 336843820513755157,
        "actor": 336843820513755157,
        "type": "watch",
        "reason": "Unwatched: Away with you!",
        "hidden": true
    }
]
```

Request with non-existing `user` as a filter `param`:
```py
r = requests.get(f"http://api.pythondiscord.local:8000/bot/infractions", params={'user': 1}, headers=auth_headers)
print(f"Status code: {r.status_code}")
print(json.dumps(r.json(), indent=4))
```
results in all fields being returned, regardless of the `user`'s id, since the invalid `user` was ignored:
```json
Status code: 200
[
    {
        "id": 1,
        "inserted_at": "2019-04-27T20:54:53.930875Z",
        "expires_at": null,
        "active": false,
        "user": 336843820513755157,
        "actor": 336843820513755157,
        "type": "watch",
        "reason": "Watch that man",
        "hidden": true
    },
    {
        "id": 2,
        "inserted_at": "2019-04-27T21:19:36.568151Z",
        "expires_at": null,
        "active": false,
        "user": 336843820513755157,
        "actor": 336843820513755157,
        "type": "watch",
        "reason": "Unwatched: Away with you!",
        "hidden": true
    }
]
```
while providing it as `user__id`, it works as expected:
```py
r = requests.get(f"http://api.pythondiscord.local:8000/bot/infractions", params={'user__id': 1}, headers=auth_headers)
print(f"Status code: {r.status_code}")
print(json.dumps(r.json(), indent=4))
```
results in no results being returned:
```json
Status code: 200
[]
```

Reply:
Maybe I didn't test it with manual requests after all. But in the tests I wrote, it does use just `user`. However, that would mean tests are not properly reproducing actual usage of the API. 

https://github.com/python-discord/site/blob/django/pydis_site/apps/api/tests/test_infractions.py#L170

Reply:
It does accept `user` when you're creating an object, since that's the `Field` name and it will convert that to a  ForeignKey to a user in the `User` table. However, when *filtering*, it takes it as is and I think it only accepts the `user__id` as a filter field, not `user` by itself.

Reply:
Good catch then. Is there anything we can do to not have the suffix though? It's kinda ugly but also not worth spending much time on.

Reply:
I don't know how to do that, but my guess is there probably is a way to do that. Maybe someone knows? Otherwise I'd have to spend some time digging through the docs/SO.

---

To be sure, I added these two tests locally to your test suite:
```py
    def test_filter_user(self):
        url = reverse('bot:infraction-list', host='api')
        response = self.client.get(
            url,
            data={
                'user': 99989890898990124
            }
        )

        self.assertEqual(response.status_code, 200)
        infractions = response.json()

        self.assertEqual(len(infractions), 0)

    def test_filter_user__id(self):
        url = reverse('bot:infraction-list', host='api')
        response = self.client.get(
            url,
            data={
                'user__id': 99989890898990124
            }
        )

        self.assertEqual(response.status_code, 200)
        infractions = response.json()

        self.assertEqual(len(infractions), 0)
```
to your test suite and the first fails:
```py
FAIL: test_filter_user (pydis_site.apps.api.tests.test_infractions.InfractionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/sebastiaan/pydis/site_django/pydis_site/apps/api/tests/test_infractions.py", line 99, in test_filter_user
    self.assertEqual(len(infractions), 0)
AssertionError: 2 != 0
```
I've also tried the `url` notation (which should be the same), `{url}?user=312498397` vs `{url}?user__id=312498397`and that gives the same result.

Reply:
I've left this as it is now. If someone has a nice way of doing this, don't hesitate to reopen this conversation.